packages/amplify-cli-core/src/extractZip.js

```diff
@@ -0,0 +1,153 @@
+    this.opts = opts;
+  }
+
+  async extract() {
+    this.zipfile = await openZip(this.zipPath, { lazyEntries: true });
+    this.canceled = false;
+
+    return new Promise((resolve, reject) => {
+      this.zipfile.on('error', (err) => {
+        this.canceled = true;
+        reject(err);
+      });
+      this.zipfile.readEntry();
+
+      this.zipfile.on('close', () => {
+        if (!this.canceled) {
+          resolve();
+        }
+      });
+
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
+      this.zipfile.on('entry', async (entry) => {
+        if (this.canceled) {
+          return;
+        }
+
+        if (entry.fileName.startsWith('__MACOSX/')) {
+          this.zipfile.readEntry();
+          return;
+        }
+
+        const destDir = path.dirname(path.join(this.opts.dir, entry.fileName));
+
+        try {
+          await fs.mkdir(destDir, { recursive: true });
+
+          const canonicalDestDir = await fs.realpath(destDir);
+          const relativeDestDir = path.relative(this.opts.dir, canonicalDestDir);
+
+          if (relativeDestDir.split(path.sep).includes('..')) {
+            throw new Error(`Out of bound path "${canonicalDestDir}" found while processing file ${entry.fileName}`);
+          }
+
+          await this.extractEntry(entry);
+          this.zipfile.readEntry();
+        } catch (err) {
+          this.canceled = true;
+          this.zipfile.close();
+          reject(err);
+        }
+      });
+    });
+  }
+
+  async extractEntry(entry) {
+    /* istanbul ignore if */
+    if (this.canceled) {
+      return;
+    }
+
+    if (this.opts.onEntry) {
+      this.opts.onEntry(entry, this.zipfile);
+    }
+
+    const dest = path.join(this.opts.dir, entry.fileName);
+
+    // convert external file attr int into a fs stat mode int
+    const mode = (entry.externalFileAttributes >> 16) & 0xffff;
+    // check if it's a symlink or dir (using stat mode constants)
+    const IFMT = 61440;
+    const IFDIR = 16384;
+    const IFLNK = 40960;
+    const symlink = (mode & IFMT) === IFLNK;
+    let isDir = (mode & IFMT) === IFDIR;
+
+    // Failsafe, borrowed from jsZip
+    if (!isDir && entry.fileName.endsWith('/')) {
+      isDir = true;
+    }
+
+    // check for windows weird way of specifying a directory
+    // https://github.com/maxogden/extract-zip/issues/13#issuecomment-154494566
+    const madeBy = entry.versionMadeBy >> 8;
+    if (!isDir) isDir = madeBy === 0 && entry.externalFileAttributes === 16;
+
+    const procMode = this.getExtractedMode(mode, isDir) & 0o777;
+
+    // always ensure folders are created
+    const destDir = isDir ? dest : path.dirname(dest);
+
+    const mkdirOptions = { recursive: true };
+    if (isDir) {
+      mkdirOptions.mode = procMode;
+    }
+    await fs.mkdir(destDir, mkdirOptions);
+    if (isDir) return;
+
+    const readStream = await promisify(this.zipfile.openReadStream.bind(this.zipfile))(entry);
+
+    if (!symlink) {
```

Comment:
might be worth:
1. leaving a comment here explaining delta between this version and library.
2. logging symlink presence?

Reply:
1. Added
2. That can be a fast-followup